### PR TITLE
Change snap packaging to build from source

### PR DESCRIPTION
**Pull request by mhall119**
## Overview

Rather than creating a snap package from pre-build binaries, this allows building the snap package directly from source. Makes it possible to use external build automation such as build.snapcraft.io 

## Testing recommendations

On any Ubuntu 16.04 or newer system, apt install "snapcraft" and then run "snapcraft snap" or "make snap". The resulting .snap file can be installed with "snap install --dangerous *.snap"

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,6 @@ config.erl
 install.mk
 rel/*.config
 rel/dev*
-rel/snap*
```
Comment (wohali) [inline .gitignore]: You probably need to add *.snap now, though.

Comment (rnewson) [inline .gitignore]: ooh I was too hasty. @mhall119 can file another PR if that's the case.

File: snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          - libnspr4-dev
+          - libicu-dev
+          - icu-devtools
+          # For erlang/rebar processing
+          - erlang-nox
+          - erlang-os-mon
+          - erlang-syntax-tools
+          # For fauxton
+          - nodejs-dev
+          - nodejs-legacy
```
Comment (kxepal) [inline snapcraft.yaml]: Wonder why two nodejs here.

File: .gitignore (hunk 1 above)
Comment (wohali) [inline .gitignore]: You also need to add `rel/snapcraft.yaml`.

Comment (mhall119) [inline .gitignore]: That hasn't changed from before, but it's easy enough to add if that's desirable

File: snapcraft.yaml (hunk 2 above)
Comment (mhall119) [inline snapcraft.yaml]: nodejs-legacy is needed for the 'node' executable, which IIRC is being called from somewhere in the build process

Comment (kxepal) [inline snapcraft.yaml]: Thanks!

**Pull request merged**